pkg/controller/deployment/util/deployment_util.go

```diff
@@ -132,6 +132,18 @@ func GetDeploymentCondition(status extensions.DeploymentStatus, condType extensi
 	return nil
 }
 
+// TODO: remove the duplicate
+// GetDeploymentConditionInternal returns the condition with the provided type.
+func GetDeploymentConditionInternal(status internalextensions.DeploymentStatus, condType internalextensions.DeploymentConditionType) *internalextensions.DeploymentCondition {
```

Comment:
If kubectl switch to versioned clientset, it will not be able to talk to api servers supporting different API versions at the same time. Not sure how much we value the version compatibility of kubectl. I can open an issue to discuss.